src/mscorlib/shared/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -781,7 +781,19 @@ private static unsafe void WriteToBuffer(byte *buffer, uint bufferLength, ref ui
             Debug.Assert(bufferLength >= (offset + 8));
             *(long *)(buffer + offset) = value;
             offset += 8;
-        }           
+        }
+
+        private static TypeCode GetTypeCodeExtended(Type parameterType)
+        {
+            // Guid is not part of TypeCode (yet), we decided to use 17 to represent it, as it's the "free slot" 
```

Comment:
FWIW, adding Guid value to the public TypeCode enum and make Type.GetTypeCode to return it would be unacceptable breaking change.

Reply:
@jkotas thanks for letting me know. Would it be a breaking change even if we add it as a not previously used value?

Reply:
The breaking change is changing `Type.GetTypeCode` to return new value.

Adding the new value to the enum without changing `Type.GetTypeCode` to return it would be technically fine, but I do not think it would make sense.

Reply:
@jkotas Thanks for the explanation, now I get it. All the `else` and `default:` `throw new NotSupported()` blocks would be executed for the new value ;)

Reply:
@jkotas, do you see TypeCode changing at all in the future to add more values?  If not, then we can essentially do this as an internal change to EventSource.  If you think that TypeCode might change in the future, then I'd be worried that this would break in the future.  Presumably, you don't think it will change since it sounds like we should not attempt to modify TypeCode to include Guid?

Reply:
Right, I do not think that TypeCode will ever change.

Reply:
Thanks @jkotas.  @adamsitnik, considering this, let's make this change internal to EventSource.